controllers/component_build_controller_common.go

```diff
@@ -68,9 +68,10 @@ func (r *ComponentBuildReconciler) GetPipelineForComponent(ctx context.Context,
 		if err != nil {
 			return nil, nil, err
 		}
-		if pipelineRef != nil {
-			return pipelineRef, pipelineParams, nil
+		if pipelineRef == nil {
+			return nil, nil, boerrors.NewBuildOpError(boerrors.ENoPipelineIsSelected, nil)
 		}
+		return pipelineRef, pipelineParams, nil
 	}
 
 	// Fallback to the default pipeline
```

Comment:
this fallback should be removed, maybe replaced by error message that no pipeline selector defined.

Reply:
Updated.